File: backend/app/schemas/financeiro.py
from datetime import date, datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel


class LancamentoCreate(BaseModel):
    consulta_id: int | None = None
    paciente_id: int
    medico_id: int | None = None
    data: date
    servico: str
    convenio: str | None = None
    valor: Decimal
    forma_pagamento: str | None = None
    observacao: str | None = None


class LancamentoUpdate(BaseModel):
    status: Literal["pago", "pendente", "atrasado"] | None = None
    forma_pagamento: str | None = None
    observacao: str | None = None
    valor: Decimal | None = None


class LancamentoResponse(BaseModel):
    id: int
    consulta_id: int | None
    paciente_id: int
    medico_id: int | None
    data: date
    servico: str
    convenio: str | None
    valor: Decimal
    status: str
    forma_pagamento: str | None
    observacao: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class IndicadoresFinanceiros(BaseModel):
    receita_paga: Decimal
    a_receber: Decimal
    atrasado: Decimal
    total_lancado: Decimal

    model_config = {
        "json_schema_extra": {
            "example": {
                "receita_paga": 1500.00,
                "a_receber": 600.00,
                "atrasado": 400.00,
                "total_lancado": 2500.00,
            }
        }
    }

File: frontend/src/types/financeiro.ts
export type LancamentoStatus = 'pago' | 'pendente' | 'atrasado';

export interface Lancamento {
  id: number;
  consulta_id: number | null;
  paciente_id: number;
  medico_id: number | null;
  data: string;
  servico: string;
  convenio: string | null;
  valor: number;
  status: LancamentoStatus;
  forma_pagamento: string | null;
  observacao: string | null;
  created_at: string;
}

export interface LancamentoCreate {
  consulta_id?: number;
  paciente_id: number;
  medico_id?: number;
  data: string;
  servico: string;
  convenio?: string;
  valor: number;
  forma_pagamento?: string;
  observacao?: string;
}

export interface LancamentoUpdate {
  status?: LancamentoStatus;
  forma_pagamento?: string;
  observacao?: string;
  valor?: number;
}

export interface IndicadoresFinanceiros {
  receita_paga: number;
  a_receber: number;
  atrasado: number;
  total_lancado: number;
}
